cmd/boost/deal_cmd.go

```diff
@@ -0,0 +1,344 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	clinode "github.com/filecoin-project/boost/cli/node"
+	"github.com/filecoin-project/boost/storagemarket/types"
+	"github.com/mitchellh/go-homedir"
+
+	types2 "github.com/filecoin-project/boost/transport/types"
+	"github.com/filecoin-project/go-address"
+	cborutil "github.com/filecoin-project/go-cbor-util"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/api"
+	chain_types "github.com/filecoin-project/lotus/chain/types"
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/google/uuid"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+	cli "github.com/urfave/cli/v2"
+
+	inet "github.com/libp2p/go-libp2p-core/network"
+)
+
+const DealProtocolv120 = "/fil/storage/mk/1.2.0"
+
+var dealCmd = &cli.Command{
+	Name:  "deal",
+	Usage: "Make an online deal with Boost",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "repo",
+			Usage: "repo directory for Boost client",
+			Value: "~/.boost-client",
+		},
+		&cli.StringFlag{
+			Name:     "http-url",
+			Usage:    "http url to CAR file",
+			Required: true,
+		},
+		&cli.StringSliceFlag{
+			Name:  "http-headers",
+			Usage: "http headers to be passed with the request",
+		},
+		&cli.StringFlag{
+			Name:     "provider",
+			Usage:    "provider address",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "commp",
+			Usage:    "commp of the CAR file",
+			Required: true,
+		},
+		&cli.Uint64Flag{
+			Name:     "piece-size",
+			Usage:    "size of the CAR file as a padded piece",
+			Required: true,
+		},
+		&cli.Uint64Flag{
+			Name:     "car-size",
+			Usage:    "size of the CAR file",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "payload-cid",
+			Usage:    "root CID of the CAR file",
+			Required: true,
+		},
+		&cli.IntFlag{
+			Name:  "start-epoch",
+			Usage: "",
+		},
+		&cli.IntFlag{
+			Name:  "duration",
+			Usage: "epochs for this deal to be active for",
+			Value: 518400, // default is 2880 * 180 == 180 days
+		},
+		&cli.IntFlag{
+			Name:  "provider-collateral",
+			Usage: "provider collateral required from storage miner",
```

Comment:
```suggestion
			Usage: "deal collateral that storage miner must put in escrow",
```

I suggest we make this a string flag and use the ParseFIL method that lotus uses.